File: frontend/features/estoque/types/requisicao.ts
export interface RequisicaoFilters {
  item_id?: number;
  data_inicio?: string;
  data_fim?: string;
  status?: StatusRequisicao;
  numero?: number;
  serie?: string;
}
export type StatusRequisicao = 'ABERTA' | 'ATENDIDA' | 'PARCIAL' | 'CANCELADA';

export interface RequisicaoItem {
  id?: number;
  item_id: number;
  embalagem_id: number;
  quantidade: number;
  atendida?: number;
}

export interface Requisicao {
  id: number;
  numero: string;
  serie?: string;
  solicitante: string;
  data_requisicao: string;
  status: StatusRequisicao;
  local_id?: number;
  itens: RequisicaoItem[];
}

export interface CreateRequisicaoDTO {
  solicitante: string;
  local_id?: number;
  itens: RequisicaoItem[];
}

export interface UpdateRequisicaoDTO {
  solicitante?: string;
  local_id?: number;
  status?: StatusRequisicao;
  itens?: RequisicaoItem[];
}

File: backend/app/modules/estoque/schemas/requisicao.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime
import enum

class StatusRequisicaoEnum(str, enum.Enum):
    ABERTA = "ABERTA"
    ATENDIDA = "ATENDIDA"
    PARCIAL = "PARCIAL"
    CANCELADA = "CANCELADA"

class RequisicaoItemBase(BaseModel):
    item_id: int
    embalagem_id: int
    quantidade: int
    atendida: int = 0

class RequisicaoItemCreate(RequisicaoItemBase):
    pass

class RequisicaoItem(RequisicaoItemBase):
    id: int
    class Config:
        orm_mode = True

class RequisicaoBase(BaseModel):
    numero: str
    serie: Optional[str] = None
    solicitante: str
    local_id: Optional[int] = None

class RequisicaoCreate(BaseModel):
    solicitante: str
    local_id: Optional[int] = None
    itens: List[RequisicaoItemCreate]

class RequisicaoUpdate(RequisicaoBase):
    status: Optional[StatusRequisicaoEnum] = None
    itens: Optional[List[RequisicaoItemCreate]] = None

class Requisicao(RequisicaoBase):
    id: int
    data_requisicao: datetime
    status: StatusRequisicaoEnum
    itens: List[RequisicaoItem]
    class Config:
        orm_mode = True
